diff --git a/java/build.gradle.kts b/java/build.gradle.kts
index 7693c66..d3941ac 100644
--- a/java/build.gradle.kts
+++ b/java/build.gradle.kts
@@ -15,7 +15,7 @@ allprojects {
 
         spotless {
             java {
-                googleJavaFormat()
+                googleJavaFormat("1.15.0")
             }
         }
 
diff --git a/java/dependencyManagement/build.gradle.kts b/java/dependencyManagement/build.gradle.kts
index 7b07f56..ab62739 100644
--- a/java/dependencyManagement/build.gradle.kts
+++ b/java/dependencyManagement/build.gradle.kts
@@ -9,7 +9,7 @@ plugins {
 data class DependencySet(val group: String, val version: String, val modules: List<String>)
 
 val DEPENDENCY_BOMS = listOf(
-    "io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom-alpha:2.16.0-alpha",
+    "io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom-alpha:2.10.0-alpha",
     "org.apache.logging.log4j:log4j-bom:2.24.3",
     "software.amazon.awssdk:bom:2.31.45"
 )
@@ -18,9 +18,9 @@ val DEPENDENCIES = listOf(
     "com.amazonaws:aws-lambda-java-core:1.2.3",
     "com.amazonaws:aws-lambda-java-events:3.15.0",
     "com.squareup.okhttp3:okhttp:4.12.0",
-    "io.opentelemetry.javaagent:opentelemetry-javaagent:2.16.0",
+    "io.opentelemetry.javaagent:opentelemetry-javaagent:2.10.0",
     "io.opentelemetry:opentelemetry-sdk-extension-aws:1.19.0",
-    "io.opentelemetry.contrib:opentelemetry-aws-resources:1.46.0-alpha",
+    "io.opentelemetry.contrib:opentelemetry-aws-resources:1.42.0-alpha",
 )
 
 javaPlatform {
