diff --git a/java/build.gradle.kts b/java/build.gradle.kts
index 3e862e7..d3941ac 100644
--- a/java/build.gradle.kts
+++ b/java/build.gradle.kts
@@ -15,7 +15,7 @@ allprojects {
 
         spotless {
             java {
-                googleJavaFormat("1.9")
+                googleJavaFormat("1.15.0")
             }
         }
 
diff --git a/java/dependencyManagement/build.gradle.kts b/java/dependencyManagement/build.gradle.kts
index b0ec233..5c2f919 100644
--- a/java/dependencyManagement/build.gradle.kts
+++ b/java/dependencyManagement/build.gradle.kts
@@ -9,7 +9,7 @@ plugins {
 data class DependencySet(val group: String, val version: String, val modules: List<String>)
 
 val DEPENDENCY_BOMS = listOf(
-    "io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom-alpha:1.30.0-alpha",
+    "io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom-alpha:1.30.0-adot-lambda1-alpha",
     "org.apache.logging.log4j:log4j-bom:2.20.0",
     "software.amazon.awssdk:bom:2.20.69"
 )
@@ -18,7 +18,7 @@ val DEPENDENCIES = listOf(
     "com.amazonaws:aws-lambda-java-core:1.2.2",
     "com.amazonaws:aws-lambda-java-events:3.11.2",
     "com.squareup.okhttp3:okhttp:4.11.0",
-    "io.opentelemetry.javaagent:opentelemetry-javaagent:1.30.0"
+    "io.opentelemetry.javaagent:opentelemetry-javaagent:1.30.0-adot-lambda1"
 )
 
 javaPlatform {
