diff --git a/collector/internal/collector/collector.go b/collector/internal/collector/collector.go
index 7cdea69..553e649 100644
--- a/collector/internal/collector/collector.go
+++ b/collector/internal/collector/collector.go
@@ -26,6 +26,7 @@ import (
 	"go.opentelemetry.io/collector/confmap/provider/envprovider"
 	"go.opentelemetry.io/collector/confmap/provider/fileprovider"
 	"go.opentelemetry.io/collector/confmap/provider/httpprovider"
+	"go.opentelemetry.io/collector/confmap/provider/httpsprovider"
 	"go.opentelemetry.io/collector/confmap/provider/yamlprovider"
 	"go.opentelemetry.io/collector/otelcol"
 	"go.uber.org/zap"
@@ -57,7 +58,14 @@ func getConfig(logger *zap.Logger) string {

 func NewCollector(logger *zap.Logger, factories otelcol.Factories, version string) *Collector {
 	l := logger.Named("NewCollector")
-	providers := []confmap.Provider{fileprovider.New(), envprovider.New(), yamlprovider.New(), httpprovider.New(), s3provider.New()}
+	providers := []confmap.Provider{
+		fileprovider.New(),
+		envprovider.New(),
+		yamlprovider.New(),
+		httpprovider.New(),
+		s3provider.New(),
+		httpsprovider.New(),
+	}
 	mapProvider := make(map[string]confmap.Provider, len(providers))

 	for _, provider := range providers {
@@ -89,8 +97,8 @@ func NewCollector(logger *zap.Logger, factories otelcol.Factories, version strin
 func (c *Collector) Start(ctx context.Context) error {
 	params := otelcol.CollectorSettings{
 		BuildInfo: component.BuildInfo{
-			Command:     "otelcol-lambda",
-			Description: "Lambda Collector",
+			Command:     "aws-otel-lambda",
+			Description: "ADOT Lambda Collector",
 			Version:     c.version,
 		},
 		ConfigProvider: c.configProvider,
