diff --git a/collector/internal/collector/collector.go b/collector/internal/collector/collector.go
index 4dcbde3..722e304 100644
--- a/collector/internal/collector/collector.go
+++ b/collector/internal/collector/collector.go
@@ -17,6 +17,7 @@ package collector
 import (
 	"context"
 	"fmt"
+	"go.opentelemetry.io/collector/confmap/provider/httpsprovider"
 	"os"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/confmap/provider/s3provider"
@@ -60,7 +61,7 @@ func NewCollector(logger *zap.Logger, factories otelcol.Factories, version strin
 	cfgSet := otelcol.ConfigProviderSettings{
 		ResolverSettings: confmap.ResolverSettings{
 			URIs:              []string{getConfig(l)},
-			ProviderFactories: []confmap.ProviderFactory{fileprovider.NewFactory(), envprovider.NewFactory(), yamlprovider.NewFactory(), httpprovider.NewFactory(), s3provider.NewFactory()},
+			ProviderFactories: []confmap.ProviderFactory{fileprovider.NewFactory(), envprovider.NewFactory(), yamlprovider.NewFactory(), httpprovider.NewFactory(), httpsprovider.NewFactory(), s3provider.NewFactory()},
 			ConverterFactories: []confmap.ConverterFactory{
 				expandconverter.NewFactory(),
 				confmap.NewConverterFactory(func(set confmap.ConverterSettings) confmap.Converter {
@@ -82,8 +83,8 @@ func NewCollector(logger *zap.Logger, factories otelcol.Factories, version strin
 func (c *Collector) Start(ctx context.Context) error {
 	params := otelcol.CollectorSettings{
 		BuildInfo: component.BuildInfo{
-			Command:     "otelcol-lambda",
-			Description: "Lambda Collector",
+			Command:     "aws-otel-lambda",
+			Description: "ADOT Lambda Collector",
 			Version:     c.version,
 		},
 		ConfigProviderSettings: c.cfgProSet,
