diff --git a/collector/internal/collector/collector.go b/collector/internal/collector/collector.go
index 4dcbde3..fc5bf0c 100644
--- a/collector/internal/collector/collector.go
+++ b/collector/internal/collector/collector.go
@@ -82,8 +82,8 @@ func NewCollector(logger *zap.Logger, factories otelcol.Factories, version strin
 func (c *Collector) Start(ctx context.Context) error {
 	params := otelcol.CollectorSettings{
 		BuildInfo: component.BuildInfo{
-			Command:     "otelcol-lambda",
-			Description: "Lambda Collector",
+			Command:     "aws-otel-lambda",
+			Description: "ADOT Lambda Collector",
 			Version:     c.version,
 		},
 		ConfigProviderSettings: c.cfgProSet,
