diff --git a/collector/internal/lifecycle/manager.go b/collector/internal/lifecycle/manager.go
index 5ec1b6a..d7bf7d2 100644
--- a/collector/internal/lifecycle/manager.go
+++ b/collector/internal/lifecycle/manager.go
@@ -90,7 +90,7 @@
 		}
 	}()
 
-	factories, _ := lambdacomponents.Components(res.ExtensionID)
+	factories, _ := lambdacomponents.Components()
 	lm.collector = collector.NewCollector(logger, factories, version)
 
 	return ctx, lm
