diff --git a/collector/internal/lifecycle/manager.go b/collector/internal/lifecycle/manager.go
index 052c45f..d2b04c4 100644
--- a/collector/internal/lifecycle/manager.go
+++ b/collector/internal/lifecycle/manager.go
@@ -86,7 +86,7 @@ func NewManager(ctx context.Context, logger *zap.Logger, version string) (contex
 		listener:        listener,
 	}
 
-	factories, _ := lambdacomponents.Components(res.ExtensionID)
+	factories, _ := lambdacomponents.Components()
 	lm.collector = collector.NewCollector(logger, factories, version)
 
 	return ctx, lm
