diff --git a/dotnet/sample-apps/aws-sdk/deploy/wrapper/main.tf b/dotnet/sample-apps/aws-sdk/deploy/wrapper/main.tf
index 0e3791a..d35a47d 100644
--- a/dotnet/sample-apps/aws-sdk/deploy/wrapper/main.tf
+++ b/dotnet/sample-apps/aws-sdk/deploy/wrapper/main.tf
@@ -1,6 +1,6 @@
 module "hello-lambda-function" {
   source  = "terraform-aws-modules/lambda/aws"
-  version = "8.8.0"
+  version = "7.19.0"
 
   architectures = compact([var.architecture])
   function_name = var.name
diff --git a/go/sample-apps/aws-sdk/deploy/wrapper/main.tf b/go/sample-apps/aws-sdk/deploy/wrapper/main.tf
index 40b69f4..5d11bce 100644
--- a/go/sample-apps/aws-sdk/deploy/wrapper/main.tf
+++ b/go/sample-apps/aws-sdk/deploy/wrapper/main.tf
@@ -1,6 +1,6 @@
 module "hello-lambda-function" {
   source  = "terraform-aws-modules/lambda/aws"
-  version = "8.8.0"
+  version = "7.19.0"
 
   architectures = compact([var.architecture])
   function_name = var.name
diff --git a/java/sample-apps/aws-sdk/deploy/agent/main.tf b/java/sample-apps/aws-sdk/deploy/agent/main.tf
index 581d026..3c267a9 100644
--- a/java/sample-apps/aws-sdk/deploy/agent/main.tf
+++ b/java/sample-apps/aws-sdk/deploy/agent/main.tf
@@ -1,6 +1,6 @@
 module "hello-lambda-function" {
   source  = "terraform-aws-modules/lambda/aws"
-  version = "8.8.0"
+  version = "7.19.0"
 
   architectures = compact([var.architecture])
   function_name = var.name
diff --git a/java/sample-apps/aws-sdk/deploy/wrapper/main.tf b/java/sample-apps/aws-sdk/deploy/wrapper/main.tf
index 3feb859..2e9e522 100644
--- a/java/sample-apps/aws-sdk/deploy/wrapper/main.tf
+++ b/java/sample-apps/aws-sdk/deploy/wrapper/main.tf
@@ -1,6 +1,6 @@
 module "hello-lambda-function" {
   source  = "terraform-aws-modules/lambda/aws"
-  version = "8.8.0"
+  version = "7.19.0"
 
   architectures = compact([var.architecture])
   function_name = var.name
diff --git a/java/sample-apps/okhttp/deploy/wrapper/main.tf b/java/sample-apps/okhttp/deploy/wrapper/main.tf
index bce4437..cfcae0e 100644
--- a/java/sample-apps/okhttp/deploy/wrapper/main.tf
+++ b/java/sample-apps/okhttp/deploy/wrapper/main.tf
@@ -1,6 +1,6 @@
 module "hello-lambda-function" {
   source  = "terraform-aws-modules/lambda/aws"
-  version = "8.8.0"
+  version = "7.19.0"
 
   architectures = compact([var.architecture])
   function_name = var.name
diff --git a/java/sample-apps/sqs/deploy/agent/main.tf b/java/sample-apps/sqs/deploy/agent/main.tf
index 54aed84..56b63b8 100644
--- a/java/sample-apps/sqs/deploy/agent/main.tf
+++ b/java/sample-apps/sqs/deploy/agent/main.tf
@@ -1,6 +1,6 @@
 module "hello-lambda-function" {
   source  = "terraform-aws-modules/lambda/aws"
-  version = "8.8.0"
+  version = "7.19.0"
 
   architectures = compact([var.architecture])
   function_name = var.name
diff --git a/java/sample-apps/sqs/deploy/wrapper/main.tf b/java/sample-apps/sqs/deploy/wrapper/main.tf
index 7cc504a..c8bae2c 100644
--- a/java/sample-apps/sqs/deploy/wrapper/main.tf
+++ b/java/sample-apps/sqs/deploy/wrapper/main.tf
@@ -1,6 +1,6 @@
 module "hello-lambda-function" {
   source  = "terraform-aws-modules/lambda/aws"
-  version = "8.8.0"
+  version = "7.19.0"
 
   architectures = compact([var.architecture])
   function_name = var.name
diff --git a/nodejs/sample-apps/aws-sdk/deploy/wrapper/main.tf b/nodejs/sample-apps/aws-sdk/deploy/wrapper/main.tf
index aed215c..09d68b5 100644
--- a/nodejs/sample-apps/aws-sdk/deploy/wrapper/main.tf
+++ b/nodejs/sample-apps/aws-sdk/deploy/wrapper/main.tf
@@ -7,7 +7,7 @@ data "aws_region" "current" {}
 
 module "hello-lambda-function" {
   source  = "terraform-aws-modules/lambda/aws"
-  version = "8.8.0"
+  version = "7.19.0"
 
   architectures = compact([var.architecture])
   function_name = var.name
diff --git a/python/sample-apps/aws-sdk/deploy/wrapper/main.tf b/python/sample-apps/aws-sdk/deploy/wrapper/main.tf
index b3242c8..4a43731 100644
--- a/python/sample-apps/aws-sdk/deploy/wrapper/main.tf
+++ b/python/sample-apps/aws-sdk/deploy/wrapper/main.tf
@@ -1,6 +1,6 @@
 module "hello-lambda-function" {
   source  = "terraform-aws-modules/lambda/aws"
-  version = "8.8.0"
+  version = "7.19.0"
 
   architectures = compact([var.architecture])
   function_name = var.name
